projects/plugins/jetpack/tests/php/modules/subscriptions/test_class.jetpack-subscriptions.php

```diff
@@ -59,20 +57,307 @@ protected function setup_jetpack_paid_newsletters( $subscription_end_date = null
+
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), '' ) );
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'everybody' ) );
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'subscribers' ) );
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'paid_subscribers' ) );
+
+		// Now we make sure we have the cookie
+		$this->assertNotNull( $_COOKIE[ WPCOM_Token_Subscription_Service::JWT_AUTH_TOKEN_COOKIE_NAME ] );
+
+		// We remove the token
+		unset( $_GET['token'] );
+
+		// We make sure everything still works
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), '' ) );
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'everybody' ) );
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'subscribers' ) );
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'paid_subscribers' ) );
+
+		// We remove the cookie
+		unset( $_COOKIE[ WPCOM_Token_Subscription_Service::JWT_AUTH_TOKEN_COOKIE_NAME ] );
+
+		// We make sure everything nothing works anymore
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), '' ) );
+		$this->assertTrue( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'everybody' ) );
+		$this->assertFalse( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'subscribers' ) );
+		$this->assertFalse( $token_subscription_service->visitor_can_view_content( array( $this->plan_id ), 'paid_subscribers' ) );
+	}
+
+	public function test_publishing_post_first_time_does_not_set_do_not_send_subscription_flag() {
+		$post_id = $this->factory->post->create();
+		wp_publish_post( $post_id );
+		$this->assertEmpty( get_post_meta( $post_id, '_jetpack_dont_email_post_to_subs', true ) );
+	}
+
+	/**
+	 * A simple helpful function to get the parameters name when filling the array
+	 *
+	 * @param string      $user_id
+	 * @param bool        $logged
+	 * @param bool        $token_set
+	 * @param bool        $post_access_level
+	 * @param bool        $is_email_sent
+	 * @param bool        $does_user_access_post
+	 * @param string|null $subscription_end_date
+	 * @param string|null $status
+	 * @return array
+	 */
+	private function accessUseCase( $user_id, $logged, $token_set, $post_access_level, $is_email_sent, $does_user_access_post, $subscription_end_date = null, $status = null ) {
+		return array( $user_id, $logged, $token_set, $post_access_level, $is_email_sent, $does_user_access_post, $subscription_end_date, $status );
+	}
+
+	public function matrixAccess() {
```

Comment:
I dont know how camelcase has been able to sneak past linter

Reply:
Definitely my bad. Fixed.